src/azul/plugins/repository/dss/__init__.py

```diff
@@ -377,6 +380,19 @@ def drs_uri(self, drs_path: str) -> str:
         netloc = config.drs_domain or config.api_lambda_domain('service')
         return f'drs://{netloc}/{drs_path}'
 
+    def direct_file_url(self,
+                        file_uuid: str,
+                        file_version: Optional[str],
+                        replica: Optional[str] = None,
+                        token: Optional[str] = None
```

Comment:
While `replica` and earlier arguments are of general use and are likely to apply to all repositories, `token` is very specific to DSS. Try to inject the token into the URL at the call site. Another option might be to exclude `token` from the abstract declaration but include it in the implementation or add a protected method that has the `token` arg, while the public method doesn't.